pkgs/development/libraries/arrow-cpp/default.nix

```diff
@@ -35,6 +35,7 @@ in stdenv.mkDerivation rec {
   patches = [
     # patch to fix python-test
     ./darwin.patch
+    ./use-upstream-target-names.patch
```

Comment:
This has to do with the enabling of CMAKE_FIND_PACKAGE_PREFER_CONFIG? We need CMAKE_FIND_PACKAGE_PREFER_CONFIG for the static build? Can we upstream this patch?

Reply:
I was planning to, just didn't get around to it yet.

Reply:
Upstreaming at https://github.com/apache/arrow/pull/6480.